Automatic calculate the assembly and package version in servicing (#57158)
dotnet/runtime: docs/project/library-servicing.md

## Check if a package is generated

> If a library likely is part of the shared framework (to check for that, look into the `NetCoreAppLibrary.props` file), then there is nothing that needs to be done here.
> Otherwise, You'll need to set `<GeneratePackageOnBuild>true</GeneratePackageOnBuild>` in the source project becuase the packages are not generated by default in the servicing releases.

Reviewer: ```suggestion
If a library is packable (check for the `<IsPackable>true</IsPackable>` property) you'll need to set `<GeneratePackageOnBuild>true</GeneratePackageOnBuild>` in the source project. That is necessary as packages aren't generated by default in servicing releases.
```

Reviewer: Starting with 6.0 we have packages which contain assets that are also redistributed inside the shared framework. We don't trim these out of our packages anymore (net6.0 assets).

If a library's source project sets `<IsPackable>true</IsPackable>` a package is generated. If the library's source project doesn't set `<IsPackable>true</IsPackable>`, the library likely is part of the shared framework (to check for that, look into the `NetCoreAppLibrary.props` file). If it is, then there is nothing that needs to be done here.

## Determine ServiceVersion

When you make a change to a library & ship it during the servicing release, the `ServicingVersion` must be bumped. This property is found in the library's source project. It's also possible that the property is not in that file, in which case you'll need to add it to the library's source project and set it to 1. If the property is already present in your library's source project, just increment the servicing version by 1.

> The packages are not generated by default in servicing. You'll need to set `<GeneratePackageOnBuild>true</GeneratePackageOnBuild>` in the source project.

Reviewer: This conflicts with the message in L7. Can you please consolidate this line with L7?

Author: are we gonna remove the IsPackable from the source project in servicing ?

Reviewer: No. That property controls if a project is packable but not if a package should be created. For that we use `<GeneratePackageOnBuild>`. The `<IsPackable>` property shouldn't be touched at all in servicing.

> ## Determine PackageVersion

Reviewer: This title should probably be updated.

> When you make a change to a library & ship it during the servicing release, the `ServicingVersion` must be bumped. This property is found in the library's source project. It's also possible that the property is not in that file, in which case you'll need to add it to the library's source project and set it to 1. If the property is already present in your library's source project, just increment the servicing version by 1.

Reviewer: > ## Check if a package is generated

In the section above, can you please update the content to express that packages aren't generated during servicing by default and `<GeneratePackageOnBuild>true</GeneratePackageOnBuild>` needs to be set in the source project file?
